File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        Map<Integer, Integer> ans = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; i++) {
            int sum = target - nums[i];

            if (ans.containsKey(sum)) {
                return new int[] { i, ans.get(sum) };
            }
            ans.put(nums[i], i);
        }
        return new int[] { -1, -1 };
    }
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> ans;

        for (int i = 0; i < nums.size(); i++) {

            int sum = target - nums[i];
            if (ans.contains(sum)) {
                return {i, ans[sum]};
            }
            ans.insert({nums[i], i});
        }
        return {-1, -1};
    }
};